connection.go

```diff
@@ -158,6 +158,10 @@ type ConnectionOptions struct {
 	// MaxCloseTime controls how long we allow a connection to complete pending
 	// calls before shutting down. Only used if it is non-zero.
 	MaxCloseTime time.Duration
+
+	// SendCancelOnContextCanceled will enable sending cancel messages
+	// when a request context is canceled before receiving a response.
```

Comment:
this is a client option, but this question brings up a good point - do we need an option on the server side as well? if there were a bad client somehow spamming cancel frames it could be pretty disruptive

Reply:
Yep this option only affects callers -- the inbound was always happening. 

I'm assuming there's no bad clients out there spamming cancel messages (since that would cause a lot of error logs with "Received unexpected frame" which I haven't heard of).

However, it's safer to put the server-side behind an option too so added an option which controls both cancel processing on the server-side, and whether cancels are forwarded by relays.

I'm using a separate option since you generally will want to enable the server-side first, then client-side.